sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -244,6 +247,17 @@ private void decodeDictionaryIds(int rowId, int num, ColumnVector column,
     switch (descriptor.getType()) {
       case INT32:
       case INT64:
+      case INT96:
+        if (column.dataType() == DataTypes.TimestampType) {
+          for (int i = rowId; i < rowId + num; ++i) {
+            Binary v = dictionary.decodeToBinary(dictionaryIds.getInt(i));
```

Comment:
Added a TODO for converting the dictionary of binaries into long to make it cheaper